org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/DocumentLifeCycleHandlerTest.java

```diff
@@ -943,18 +944,19 @@ private void assertNewProblemReported(ExpectedProblemReport... expectedReports)
 		List<PublishDiagnosticsParams> diags = getClientRequests("publishDiagnostics");
 		assertEquals(expectedReports.length, diags.size());
 
-		for (int i = 0; i < expectedReports.length; i++) {
-			PublishDiagnosticsParams diag = diags.get(i);
-			ExpectedProblemReport expected = expectedReports[i];
-			assertEquals(JDTUtils.toURI(expected.cu), diag.getUri());
+		for (ExpectedProblemReport expected : expectedReports) {
```

Comment:
I found the order of the diagnostics might not be aligned, so I changed this test case.

If it's on purpose, please let me know.

Reply:
Seems fine to me. I don't think we need to assert order, so this should improve overall stability.